src/lib/models/base/image-base-model.ts

```diff
@@ -16,11 +17,17 @@ export class ImageBaseModel<
   }
 
   async generate(body: RequestType): Promise<ResponseType> {
-    const { image } = body;
-    const base64Content = fs.readFileSync(image).toString("base64");
-    const response = await this.client.post<ResponseType>({
-      ...body,
-      image: base64Content,
+    const { image , ...rest} = body;
+    const formData = new FormData();
+    const readStream = await ReadStreamUtils.getReadStream(image);
+    formData.append("image", readStream);
+    Object.entries(rest).forEach(([key, value]) => {
+      formData.append(key, JSON.stringify(value));
+    });
```

Comment:
You can probably wrap this in a utility method, that receives "blob-like" key names, and treats the rest via JSON.stringify. One reason is to reduce duplication, another is that I'm worried the JSON.stringify is not the right option, but if it's in one place it would be easier to manage.

I assume you have tested this on all affected categories with some non-blob inputs?

Reply:
Gotcha. I'm going to wrap form data serialization under one utility class.

Reply:
I tested image input from URL, from file and from buffer. Blob should also be included.